daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala

```diff
@@ -319,6 +321,15 @@ object Hash {
   ): Either[String, Hash] =
     handleError(assertHashContractKey(templateId, key))
 
+  def hashCommittedContract(
+      templateId: Ref.Identifier,
+      key: Value[Value.ContractId],
```

Comment:
key should be arg actually, right (`s/key/arg/) ? Also do we not need to include all `ContractInst` parameters, i.e. `agreementText` too?

More on the stylistic side on which I am flexible, just my 2 cents: I was assuming we'd actually pass in a `ContractInst` and not necessarily refer to the contract as committed.

Reply:
You right, `"key"` should be `"arg"`.

Reply:
I am more incline to ignore the `agreementText` as it is not semantically part of the contract and we may want to drop it later.

Reply:
I add a function for `Value.ContractInst[VersionedValue[ContractId]]`

Reply:
Re agreement text: I always thought that the agreement text is just a DAML function of the template arguments. If that's also the case in DAML-LF, then no need to add the agreement text.

Reply:
That’s correct.